### ISSUE TITLE
Issue when running dotnet publish

### ISSUE THREAD

**Issue by MarcinuB** [ncalc/ncalc-async#17]
Hi there. 

I'm trying to use that library to replace the sync version of NCalc. I use it in the app targetting .NET Core 6 and it all works fine when debugging, but when I try to publish using dotnet publish I'm getting a whole list of NU1605 errors:

 error NU1605:  <project name> -> NCalcAsync 3.2.6 -> Antlr3.Runtime 3.5.1 -> NETStandard.Library 1.6.0 -> System.Console 4.0.0 -> runtime.unix.System.Console 4.3.0 -> System.Collections (>= 4.3.0) 
error NU1605:  <project name> -> NCalcAsync 3.2.6 -> Antlr3.Runtime 3.5.1 -> NETStandard.Library 1.6.0 -> System.Console 4.0.0 -> runtime.unix.System.Console 4.3.0 -> System.IO.FileSystem.Primitives (>= 4.3.0) 

There is a long list of that errors and it all seems to be referencing the .NETStandard.Library 1.6.0.

Any idea how I can fix that?

**Comment by petli** [comment]
There was a workaround for .NET 3.1 that may still work, which is adding this package reference:

```
    <!-- 
        This is a workaround for conflicting package versions when packaging a Lambda.
        The dependency chain NCalcAsync -> Antlr.Runtime -> NETStandard.Library 1.6.0 gets precedence 
        over the .NET Core 3.1 packages when specifying the linux-x64 runtime (which dotnet lambda package
        does).  Referencing the 3.1 targets with private assets here forces the nearest-win rule to favour
        the 3.1 version, resolving the conflict.
        https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1605#example-2
    -->
    <PackageReference Include="Microsoft.NETCore.Targets" Version="3.1.0" PrivateAssets="all" />
```

However, we ran into other problems later that this didn't fix. I only know two ways to fix this:

* Port ncalc-async to ANTLR 4, which is maintained and doesn't have the problematic .NET Standard 1.6 target. ANTLR 4 seems to be pretty backward compatible in terms of the features ncalc-async uses, but it can generate visitor classes directly so there would be quite some scope for improving the grammar usage.
* Find the ANTLR 3 maintainer and get a new version of that package released that targets .NET Standard 2.0 instead.

I should add here that I didn't bother with option 1 myself, since the work project were we used ncalc-async had a custom grammar and only used the ncalc-async engine. Porting the library to ANTLR 4 in a way that still allowed us to use a custom grammar seemed tricky, and hooking the ncalc-async engine into our code was always a bit clumsy. Thus we replaced ncalc-async completely with our own ANTLR 4-based grammar and code that compiled the expressions to async lambdas.  One of our devs expressed interest in trying to pull this into a new open source library, but we just haven't had time, alas. This would be something that wouldn't be compatible with the old ncalc APIs, however, so it would be a completely new libray.
